plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/expression/ImplementMinMax.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.query.expression;
+
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.pinot.PinotColumnHandle;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.Variable;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.pinot.query.DynamicTableBuilder.SUPPORTED_INPUT_TYPES;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.expressionType;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.singleInput;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.variable;
+import static java.lang.String.format;
+
+/**
+ * Implements {@code min(x)}, {@code max(x)}.
+ */
+public class ImplementMinMax
+        implements AggregateFunctionRule
+{
+    // Extracted from io.trino.plugin.jdbc.expression.ImplementMinMax
+    private static final Capture<Variable> INPUT = newCapture();
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().matching(Set.of("min", "max")::contains))
+                .with(singleInput().matching(
+                        variable()
+                                .with(expressionType().matching(SUPPORTED_INPUT_TYPES::contains))
```

Comment:
SUPPORTED_INPUT_TYPES doesn't look specific to min or max (to comparisons).

i would suggest defining supported data types directly here, in `ImplementMinMax`